src/puptoo/process/profile.py

```diff
@@ -412,21 +412,23 @@ def system_profile(
 
     if redhat_release and os_release:
         try:
-            if redhat_release.minor is None:
-                redhat_release.minor = 0
-
-            profile["os_release"] = '{0}.{1}'.format(redhat_release.major, redhat_release.minor)
-            profile["operating_system"] = {
-                "major": redhat_release.major,
-                "minor": redhat_release.minor,
-                "name": os_release.release
-            }
-            profile["system_update_method"] = "yum"
-
-            if os_release.is_rhel:
-                if profile.get("host_type") is None:
-                    if redhat_release.major >= 8:
-                        profile["system_update_method"] = "dnf"
+
```

Comment:
I wanted to ask some questions about some of these changes. I see that there are a few commits, so I believe you're probably still testing things.  Just to make sure I didn't leave out any details, we're expecting the 'operating_system' response to either say RHEL or CentOS Linux, no other systems are supported for this work.  This code looks significantly different than the logic I submitted and I'm not able to understand why such a different approach is being used.  I'm including some screenshots from Ephemeral where the code I provided is doing exactly what we want. I just wanted to make sure we're on the same page and if there is a problem with the logic I suggested that I get the opportunity to understand the reasoning behind it.

![Screenshot from 2023-08-09 19-04-24](https://github.com/RedHatInsights/insights-puptoo/assets/23130869/f26bfbfe-6e9f-46c3-bf5e-13fd74c54752)
![Screenshot from 2023-08-09 18-59-20](https://github.com/RedHatInsights/insights-puptoo/assets/23130869/bff8f544-0b3e-4638-91b5-2926422e6884)
![Screenshot from 2023-08-09 18-58-58](https://github.com/RedHatInsights/insights-puptoo/assets/23130869/13a50052-0f79-42aa-a016-b561cbb2bb0c)
![Screenshot from 2023-08-09 18-58-18](https://github.com/RedHatInsights/insights-puptoo/assets/23130869/1c71f69d-cf96-4e21-91b1-66eb045d0bfc)

Reply:
> Just to make sure I didn't leave out any details, we're expecting the 'operating_system' response to either say RHEL or CentOS Linux, no other systems are supported for this work. 

Yes, the latest code will populate the field with either`RHEL` or `CentOS Linux`, others are rejected. You can check the tests I have added.

>  This code looks significantly different than the logic I submitted and I'm not able to understand why such a different approach is being used.

With your code, other Linux distributions like `CentOS Stream`, `Fedora` etc (other than `CentOS Linux` and `RHEL` that we currently want to support) were also passing. And insights-core version `3.2.9` has this [issue](https://github.com/RedHatInsights/insights-core/pull/3872) and there was an update on the `release` attribute of `os_release` combiner - [here](https://github.com/RedHatInsights/insights-core/pull/3879). So I have fixed these bugs. Also, there were code redundancies.

I have not squashed the commits so that you can easily check what changes I added on every commit.

Let me know if you have any doubts.